api/src/main/java/io/cloudevents/CloudEvent.java

```diff
@@ -24,9 +24,23 @@
  */
 public interface CloudEvent extends CloudEventAttributes, CloudEventExtensions {
 
+    /**
+     * The event data
+     */
+    @Nullable
+    Object getRawData();
+
     /**
      * The event data
      */
     @Nullable
     byte[] getData();
+
+    /**
+     * The event data as T
+     *
+     * @throws IllegalArgumentException if there isn't any unmarshaller from the inner data to T
+     */
+    @Nullable
+    <T> T getData(Class<T> c) throws IllegalArgumentException;
```

Comment:
as seen in Jackson and others, `Class` may not represent all the types (e.g. `List<User>`). Consider using a type reference instead.

Reply:
I thought about that, but that means we need to define our type reference? I'm not sure i wanna dive in into that complexity...

Reply:
My idea for using `Class` is:

* For the simple use cases, the user just uses `getData()`.
* For the user cases when they need a type reference, they just use jackson directly: they get the raw payload with `getData(Object.class)` and then they use the jackson mapper to map it.

Reply:
I am personally against this operation all together regardless of `Type` or `Class`. `CloudEvent` type is representation of the actual cloud event on the wire. It should be as simple and as light as possible containing only essentials. We had a very long and [contentious debate about it](https://github.com/cloudevents/sdk-java/pull/143#issuecomment-632501986), so let's not go back to square one. 
If user needs stronger type for its data representation then there should be another abstraction that would perform such conversion, but only as an option.